client-mobile/src/models/ingredientModel.ts

```diff
@@ -0,0 +1,106 @@
+import { CreateIngredientInput, Ingredient } from "@client/api/schema"
+import { databaseClient } from "@client/clients/databaseClient"
+import "react-native-get-random-values"
+import { v4 as uuidv4 } from "uuid"
+
+/**
+ * IngredientModel class provides methods for performing database operations on the ingredients table.
+ */
+export class IngredientModel {
+  static initSchema = async () =>
+    databaseClient.transactionAsync(async (tx) => {
+      await tx.executeSqlAsync(`
+        CREATE TABLE IF NOT EXISTS ingredients (
+          id TEXT PRIMARY KEY,
+          name TEXT NOT NULL,
+          unit TEXT,
+          last_category_id TEXT,
+          created_at TEXT NOT NULL DEFAULT CURRENT_TIMESTAMP
+        );
```

Comment:
We need to make sure that there are foreign keys:

```suggestion
      CREATE TABLE IF NOT EXISTS ingredients (
          id TEXT PRIMARY KEY,
          name TEXT NOT NULL CHECK(length(name) <= 100),
          unit TEXT,
          last_category_id TEXT,
          created_at TEXT NOT NULL DEFAULT CURRENT_TIMESTAMP,
          FOREIGN KEY (unit) REFERENCES units(id)
        );
```

Reply:
Additionally, please modify `useAsyncEffect` hook in `client-mobile/src/hooks/useLocalDbInit.ts`:

```
useAsyncEffect(
    async () => {
      try {
        await databaseClient.transactionAsync(async (tx) => {
          await tx.executeSqlAsync(`
            PRAGMA foreign_keys = ON;
          `)
        })

        await UnitModel.initSchema()
        await IngredientModel.initSchema()
        await ShoppingListModel.initSchema()
        
        setIsDbInitialized(true)
      } catch (error) {
        setDbInitError(error as Error)
      }
    },
    async () => {},
    [],
  )
```